core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4633,7 +4648,30 @@ private QueryModel rewriteSampleBy(@Nullable QueryModel model) throws SqlExcepti
 
                         if (timestampAlias != null) {
                             timestampColumn = tableNamePrefixedTimestampColumn;
+                        }
+                    }
+                }
 
+                if (maybeKeyed.size() > 0 && sampleByFillSize > 0) {
+                    if (!isNoneKeyword(sampleByFill.getQuick(0).token)) {
+                        for (int i = 0, n = maybeKeyed.size(); i < n; i++) {
+                            final ExpressionNode expr = maybeKeyed.getQuick(i);
+                            // drop out early, since we don't handle keyed
+                            if (!Chars.equalsIgnoreCase(expr.token, timestamp.token) && !Chars.equalsIgnoreCaseNc(expr.token, timestampAlias)
+                                    && sampleByFill.size() != 0) {
```

Comment:
We have this check under the `if (!isNoneKeyword(sampleByFill.getQuick(0).token))` branch, so if `sampleByFill.size()` is 0, the `if` would have already thrown.